plugins/contact.py

```diff
@@ -0,0 +1,101 @@
+""" A bot to make the wordkeeper's job a little easier during games of Contact.
+
+wordkeeper> !challenge @player1 @player2
+slaskbot> @player1 @player2: type !ready when here
+player1> !ready
+player2> !ready
+slaskbot> (@wordkeeper) Starting countdown...
+slaskbot> 3
+slaskbot> 2
+slaskbot> 1
+slaskbot> go!
+
+"""
+
+import logging
+import re
+import time
+
+import config
+from models.contact import Challenge
+
+
+# Number of seconds we wait for someone to be ready
+TIMEOUT = 120
+
+
+def on_message(msg, server):
+    channel = server["channel"]
+    if channel not in config.CONTACT_CHANNELS:
+        return
+
+    with Challenge.lock(channel) as challenge:
+        if re.search(r"!contactreset", msg.get("text", "")):
+            challenge.reset(channel)
+            return "contactbot reset"
+
+        state = challenge.state
+        if state == Challenge.COMPLETED:
+            return start_challenge(challenge, msg)
+        elif state == Challenge.WAITING_FOR_READY:
+            return set_ready(challenge, msg)
+        elif state > 0:
```

Comment:
Tick can't happen unless someone sends a message.
!challenge and !ready are fine, but the countdown will have issues.
Also, !ready expiring can only happen when someone in the contact list says "!ready" (unless you run the check on all messages (not !ready, not-challenged user, etc).

(for some reason the selected code isn't what I highlighted. I still need to learn how that works apparently)